eng/packages/http-client-csharp/generator/Azure.Generator/src/Visitors/ModelFactoryRenamerVisitor.cs

```diff
@@ -13,6 +13,9 @@ internal class ModelFactoryRenamerVisitor : ScmLibraryVisitor
         {
             if (type is ModelFactoryProvider && type.CustomCodeView == null)
             {
+                // Reset the type provider so that all the methods will be recomputed after the name change.
+                // This is necessary because the name change will impact the custom code view calculation.
+                type.Reset();
```

Comment:
Is this behavior specific to the visitor, or should a name update do this implicitly?

Reply:
As discussed offline, we will add a new `reset` parameter to the update method.